### PR TITLE
mptcp: initial disconnect coverage

### PR DESCRIPTION
**Pull request by pabeni**
Add basic test for connect(AF_UNSPEC) for a newly accepted
socket.

Signed-off-by: Paolo Abeni <pabeni@redhat.com>

### CODE REVIEW

File: gtests/net/mptcp/syscalls/disconnect_after_accept.pkt
Comment (matttbe) [inline gtests/net/mptcp/syscalls/disconnect_after_accept.pkt]: Small detail: do we need to specify the addresses? `addr[caddr0] > addr[saddr0]`

**Review by matttbe — COMMENTED** [review]
Thank you for sharing this!

Just one small comment, the rest looks good to me!
I can do the modif if you prefer.

**Review by dcaratti — APPROVED** [review]
LGTM!

File: gtests/net/mptcp/syscalls/disconnect_after_accept.pkt
Comment (matttbe) [inline gtests/net/mptcp/syscalls/disconnect_after_accept.pkt]: (as discussed on IRC: you switched to v0 here)

File: gtests/net/mptcp/syscalls/disconnect_after_accept.pkt
Comment (matttbe) [inline gtests/net/mptcp/syscalls/disconnect_after_accept.pkt:13]: (not sure we need mss 1000 but it doesn't hurt)

**Review by matttbe — CHANGES_REQUESTED** [review]

**Review by matttbe — APPROVED** [review]

**Comment by matttbe** [comment]
We can merge this once Paolo's kernel patch is merged in our tree.

I tested the new .pkt on top of the export branch, with and without Paolo's fix. I can reproduce the issue without the fix.

**Pull request merged**
